### ISSUE TITLE
Itertrain method to include validation set errors

### ISSUE THREAD

**Issue by dnth** [lmjohns3/theanets#63]
It would be really nice to have the itertrain method to return the validation set errors in order for a plot of learning curves.

**Comment by lmjohns3** [comment]
I've changed the training interface to generate a (train, valid) monitor pair after every iteration. The validation monitors might not be updated every training iteration, but whatever is most recent will be returned.

    for train, valid in exp.itertrain(dataset):
        print('training loss:', train['loss'])
        print('most recent validation loss:', valid['loss'])
